### PR TITLE
fix onboarding flash and progress animation

### PR DESCRIPTION
**Pull request by pvicensSpacedev**
### Summary of Changes

- Fixed onboarding flash on next step, which sort of broke the animation of the progress bar so also fixed that by adding a context for the actual progress step

### Demo or Before/After Pics

https://github.com/gallery-so/gallery/assets/112896251/fb6aa0e1-1a6d-4e97-8fc1-8d27158ded9a

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/components/Onboarding/FullPageCenteredStep.tsx [hunk 1]
```diff
@@ -49,14 +49,13 @@ const StyledPage = styled.div<{ withFooter?: boolean }>`
...
   to: number;
 };
 
-function OnboardingProgressBar({ from, to }: OnboardingProgressBarProps) {
+function OnboardingProgressBar({ to }: OnboardingProgressBarProps) {
   return (
     <StyledBar
-      initial={{ width: from }}
+      initial={{ width: `${to}%` }}
       animate={{ width: `${to}%` }}
```
Comment (Robinnnnn) [inline apps/web/src/components/Onboarding/FullPageCenteredStep.tsx:59]: why do we have both `initial` and `animate` props pointing to `to%`?

File: apps/web/src/contexts/AppProvider.tsx [hunk 2]
```diff
@@ -53,56 +54,58 @@ export default function AppProvider({
...
-                    </SwrProvider>
-                  </WebErrorReportingProvider>
-                </AnalyticsProvider>
-              </PrivyProvider>
-            </RelayEnvironmentProvider>
-          </Boundary>
-        </MaintenanceStatusProvider>
-      </ToastProvider>
-    </Boundary>
+    <ProgressProvider>
```
Comment (Robinnnnn) [inline apps/web/src/contexts/AppProvider.tsx:57]: nit: let's name this `OnboardingProgressProvider`

File: apps/web/src/contexts/onboardingProgress/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  const setProgress = useCallback((stepName: StepName) => {
+    const step = ONBOARDING_PROGRESS_BAR_STEPS[stepName];
+    if (step) {
+      setProgressState({ from: step.from, to: step.to });
+    }
+  }, []);
+
+  return (
+    <ProgressContext.Provider value={{ ...progress, setProgress }}>
```
Comment (Robinnnnn) [inline apps/web/src/contexts/onboardingProgress/index.tsx:24]: let's make sure to `useMemo` the value

**Review by Robinnnnn — APPROVED** [review]
